### PR TITLE
Fix intense CPU usage when sessionToken is invalid in liveQuery

### PR DESCRIPTION
**Pull request by flovilmart**
When sessionToken fetch failed through getAuthForSessionToken auth and userId would be undefined. 

This PR ensure we do not throw an exception in this case.

### CODE REVIEW

**Comment by dplewis** [comment]
I don’t think it’s needed but test case?

**Comment by flovilmart** [comment]
@dplewis this is a case that was overlooked initially.

File: src/LiveQuery/ParseLiveQueryServer.js [hunk 1]
```diff
@@ -527,27 +531,40 @@ class ParseLiveQueryServer {
...
       })
-      .then(async isRoleMatched => {
-        if (isRoleMatched) {
-          return Promise.resolve(true);
-        }
-
-        // Check client sessionToken matches ACL
-        const clientSessionToken = client.sessionToken;
-        if (clientSessionToken) {
-          const { userId } = await this.getAuthForSessionToken(
```
Comment (flovilmart) [inline src/LiveQuery/ParseLiveQueryServer.js]: this was initally overlooked, the new implementation takes properly care of both tokens

**Comment by flovilmart** [comment]
@westhom this is the fix you are looking for. Probalby the behaviour will require tweaking. Invalid session tokens should not issue server calls for 10 minutes (they may call after) and you should see performance improvements at large.

**Comment by flovilmart** [comment]
@dplewis @acinader what do you think of this implementaton?

File: src/LiveQuery/ParseLiveQueryServer.js [hunk 2]
```diff
@@ -482,25 +492,19 @@ class ParseLiveQueryServer {
...
       return false;
     }
 
-    // TODO: get auth there and de-duplicate code below to work with the same Auth obj.
-    const { auth, userId } = await this.getAuthForSessionToken(
-      subscriptionInfo.sessionToken
-    );
+    const { auth, userId } = await this.getAuthForSessionToken(token);
+
+    // Getting the session token failed
```
Comment (acinader) [inline src/LiveQuery/ParseLiveQueryServer.js]: fail fast

**Pull request merged**